google/internal/oauth/stsExchange.go

```diff
@@ -0,0 +1,83 @@
+package externalaccount
+
+import (
+	"context"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"strconv"
+	"strings"
+)
+
+func ExchangeToken(ctx context.Context, endpoint string, request *STSTokenExchangeRequest, authentication ClientAuthentication, headers http.Header, options map[string]interface{}) (*STSTokenExchangeResponse, error) {
+
+	client := &http.Client{}
+
+	data := url.Values{}
+	data.Set("audience", request.Audience)
+	data.Set("grant_type", "urn:ietf:params:oauth:grant-type:token-exchange")
+	data.Set("requested_token_type", "urn:ietf:params:oauth:token-type:access_token")
+	data.Set("subject_token_type", request.SubjectTokenType)
+	data.Set("subject_token", request.SubjectToken)
+	data.Set("scope", strings.Join(request.Scope, " "))
+
+	req, err := http.NewRequest("POST", endpoint, strings.NewReader(data.Encode()))
+	if err != nil {
+		fmt.Errorf("oauth2/google: failed to properly build http request")
+	}
+	for key, _ := range headers {
+		for _, val := range headers.Values(key) {
+			req.Header.Add(key, val)
+		}
+	}
+	if authentication.ClientID != "" && authentication.ClientSecret != "" {
+		plainHeader := authentication.ClientID + ":" + authentication.ClientSecret
+		req.Header.Add("Authorization", "Basic "+base64.StdEncoding.EncodeToString([]byte(plainHeader)))
+	}
+	req.Header.Add("Content-Length", strconv.Itoa(len(data.Encode())))
+
+	fmt.Printf("Transmitted request: %+v\n\n", req)
```

Comment:
Debug strings shouldn't be included in production code.

Reply:
Hey- this was mostly a dummy draft PR so that my mentor could review this before I officially submitted it to Gerrit.  I didn't realize at the time that it would be automatically forwarded to Gerrit; my bad.  I'll close this and re-open another PR with a final version later.

One question- I read the guidelines in Hey- this was mostly a dummy draft PR so that my mentor could review this before I officially submitted it to Gerrit.  I didn't realize at the time that it would be automatically forwarded to Gerrit; my bad.  I'll close this and re-open another PR with a final version later.

One question- I read the guidelines in https://golang.org/doc/contribute.html and it said that I needed to work through a Google account; this GitHub account doesn't use a Google email address.  Should I switch to that one for any future PR's?  I think I've worked out how to operate two Git accounts on one computer semi-smoothly.

(Also, where should I be replying to comments?  I tried looking on Gerrit but couldn't find anywhere- I suspect I'm not supposed to use GitHub but I couldn't find anything else)

Reply:
> Should I switch to that one for any future PR's?
I think you should stick with the corp account. There might be legal (I'm not sure actually) issues using your personal account.